### PR TITLE
[BUGFIX release] `willRender` is called for `attributeBindings` changes.

### PR DESCRIPTION
**Pull request by rwjblue**
Depending on which node-manager was used, `scope.view` or `scope.component` might be used (but not both). For `attributeBindings` case `scope.component` was not set, therefore the various hooks were not being called properly.

### CODE REVIEW

File: packages/ember-htmlbars/tests/integration/component_lifecycle_test.js [hunk 1]
```diff
@@ -344,6 +302,51 @@ QUnit.test('changing a component\'s displayed properties inside didInsertElement
...
+    })
+  }));
+
+  registry.register('template:components/the-top', compile('Top'));
+
+  view = EmberView.extend({
+    template: compile('{{the-top}}'),
+    container: container
+  }).create();
+
```
Comment (tomdale) [inline packages/ember-htmlbars/tests/integration/component_lifecycle_test.js]: This test is testing components, so why is `scope.component` not set? Surely it goes through the `ComponentNodeManager`? Am I just misunderstanding the test?

Comment (rwjblue) [inline packages/ember-htmlbars/tests/integration/component_lifecycle_test.js]: I agree that this seems somewhat fishy.  It is possible that this fix is still correct, but I'd like to figure out why `scope.component` is not set before we decide to pull the trigger on it.

**Comment by jmurphyau** [comment]
Does this fix #11533? I know its already fixed with another PR but this I think could have also fixed it

**Comment by rwjblue** [comment]
@jmurphyau - Yes, that is exactly what made me start thinking about this.

**Comment by rwjblue** [comment]
@tomdale - I tracked down why `scope.component` is not set:

* `scope.component` should be set in `bind-shadow-scope`, but is only set when `options.attrs` and `options.view` are both present. [See here](https://github.com/emberjs/ember.js/blob/dc479cef91c1b9815157612c2a2f9483a2a04edc/packages/ember-htmlbars/lib/hooks/bind-shadow-scope.js#L34)
* But for the components template, we are using our friend `manualElement` and are only passing `view` into the `blockFor` invocation. [See here](https://github.com/emberjs/ember.js/blob/dc479cef91c1b9815157612c2a2f9483a2a04edc/packages/ember-views/lib/system/build-component-template.js#L92-L98)

So it isn't possible that `scope.component` will be set for the components own element (because we were never passing `attrs` in that scenario).

I have updated the code in `bind-shadown-scope` to set `scope.component` if the `view` is stamped with `view.isComponent`.

**Pull request closed without merging**